src/Verticals/Client.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+namespace Automattic\WooCommerce\Blocks\Verticals;
+
+/**
+ * Verticals API client.
+ */
+class Client {
+	const ENDPOINT = 'https://public-api.wordpress.com/wpcom/v2/site-verticals';
+
+	/**
+	 * Make a request to the Verticals API.
+	 *
+	 * @param string $url The endpoint URL.
+	 *
+	 * @return array|\WP_Error The response body, or WP_Error if the request failed.
+	 */
+	private function request( string $url ) {
+		$response = wp_remote_get( $url );
+		if ( is_wp_error( $response ) ) {
+			return $response;
```

Comment:
Similar to https://github.com/woocommerce/woocommerce-blocks/pull/10687#discussion_r1305825414 : how about making the error msg here more specific & translatable?